cloud/aws/actuators/machine/actuator.go

```diff
@@ -142,7 +142,19 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 // Update updates a machine and is invoked by the Machine Controller
 func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	glog.Infof("Updating machine %v for cluster %v.", machine.Name, cluster.Name)
-	return fmt.Errorf("TODO: Not yet implemented")
+
```

Comment:
I think this approach to Update is sufficient to start, but it will probably be a good idea to add a comment to state that mutating the machine based on changes to the config, or handling machine upgrades is still an outstanding todo.

Reply:
Would a full update of the machine object be the thing to do here, then to update the entire thing at once? Something that eventually calls `s.Machine.Update(machine)` in the style of this? https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cloud/aws/services/cluster/service.go#L25

Reply:
I believe there are few different use cases that will need to be taken into account and we would need to decide how we want to handle them as part of the implementation:

- User tries to mutate attributes of the machine that cannot be done in place
  - either fail to update, bubbling up appropriate error message to Machine object
  - or potentially delete and recreate the machine
    - would need to take into account if the machine is a control plane node or not
    - would need to be able to attempt to gracefully cordon/drain the node prior to removing the existing machine.
    - would need to take into account if/how to recover in the case of failure
- User tries to mutate attributes of the machine that can be done in place, such as instance size
  - attempt to gracefully cordon/drain the node prior to shutting the instance down and mutating the instance before restarting and uncordoning.
  - this would also need to take into account if/how to recover in the case of failure
- User attempts to perform an upgrade on the machine
  - the workflow(s) for this would need to be designed, scoped, and implemented